pulsar-common/src/main/java/org/apache/pulsar/common/nar/NarUnpacker.java

```diff
@@ -75,6 +79,7 @@ static File doUnpackNar(final File nar, final File baseWorkingDirectory, Runnabl
                 throw new IOException("Cannot create " + parentDirectory);
             }
         }
+        setPosixFilePermissions(parentDirectory,"rwxr-x---");
```

Comment:
This only need to set after `mkdirs`. 
It's better to create a new method like `makeDirs(File)` and set permission there.